GameServerLib/GameObjects/AttackableUnits/AI/Minion.cs

```diff
@@ -25,18 +28,34 @@ public class Minion : ObjAiBase, IMinion
             uint netId = 0
         ) : base(game, model, new Stats.Stats(), 40, x, y, visionRadius, netId)
         {
-            if (!(owner == null) && owner is IChampion)
+            Name = name;
+
+            Owner = owner;
+
+            if (!(Owner == null) && Owner is IChampion)
             {
-                SetTeam(owner.Team);
+                SetTeam(Owner.Team);
                 IsPet = true;
+                if (model == Owner.Model) // Placeholder, should be changed
+                {
+                    IsClone = true;
+                }
             }
-            else {
+            else
+            {
+                SetTeam(Team);
                 IsPet = false;
             }
 
-            Owner = owner;
+            if (this is ILaneMinion)
+            {
+                IsLaneMinion = true;
```

Comment:
Consider put this in `LaneMinion` constructor.

Reply:
I am going to delete LaneMinion altogether, that's fine right?

Reply:
Why? We at the end accepted that it should be inherited. Just change how you use the base class.

Reply:
> yeah, probably delete this class/interface. Just bool variable in minion class.

Is what you said in response to:

> I think the idea of having LaneMinion be seperate from Minion is not accurate to league anyway.

Am I misinterpreting it?

Reply:
At the end we concluded that you should remove this bool and just use Minion class in the right way. Like sending the 3 packet instead of Minion packet.

Reply:
And as we concluded later, Also IsLaneMinion flag is ok.

Reply:
`IsLaneMinion` is league concept which we dont yet know how it is actually used. I consider this as solved but opened issue to understand what `IsLaneMinion` flag is for. Opened issue  #797